actions/generate_changelog_test.go

```diff
@@ -0,0 +1,184 @@
+package actions
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-github/github"
+)
+
+var (
+	// mux is the HTTP request multiplexer used with the test server.
+	mux *http.ServeMux
+
+	// client is the GitHub client being tested.
+	client *github.Client
+
+	// server is a test HTTP server used to provide mock API responses.
+	server *httptest.Server
+)
+
+// setup sets up a test HTTP server along with a github.Client that is
+// configured to talk to that test server.  Tests should register handlers on
+// mux which provide mock responses for the API method being tested.
+func setup() {
+	// test server
+	mux = http.NewServeMux()
+	server = httptest.NewServer(mux)
+
+	// github client configured to use test server
+	client = github.NewClient(nil)
+	url, _ := url.Parse(server.URL)
+	client.BaseURL = url
+	client.UploadURL = url
+}
```

Comment:
may be easier to have this function create a new `*http.ServeMux`, `*github.Client` and `*httptest.Server` for each new test, so that future tests that need to execute concurrently won't have issues.

also, noticed that the below tests call `teardown`, which results in `server.Close()` being called multiple times, and that's not guaranteed not to panic.